Assignment 3.Rmd

```diff
@@ -127,29 +146,192 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+
+for (i in 3:8){
+  DF_class[,i] <- gsub( " ", "", DF_class[,i]) 
+}
+
+DF_class[DF_class=="4050"] <- "HUDK4050"
+DF_class[DF_class=="4125"] <- "HUDM4125"
+DF_class[DF_class=="QMSS-5072"] <- "QMSS5072"
+DF_class[DF_class=="5126"] <- "HUDM5126"
+DF_class[DF_class== c("G5067", "QMSSGR5067")] <- "QMSS5067"
+DF_class[DF_class=="G5072"] <- "QMSS5072"
+DF_class[DF_class=="HUDK4050"] <- ""
+
+
+DF_class <- gather(DF_class, ClassNo, Classes, 3:8)
+DF_class <- DF_class[order(DF_class$Name),]
+
+DF_class <- DF_class[!(DF_class$Classes == ""), ] %>%
+  dplyr::select(Name, Classes)
+
+DF_class$attend <- 1
+
+DF_class <- DF_class %>%
+  spread(Classes, attend) 
+
+DF_class[is.na(DF_class)] <- 0
+
+DF_class$Name <- toupper(DF_class$Name)
+
+rownames(DF_class) <- DF_class$Name
+
+DF_class <- DF_class %>%
+  dplyr::select(-Name)
+
+DF_class_matrix <- as.matrix(DF_class)
+adj_matrix <- DF_class_matrix %*% t(DF_class_matrix)
+diag(adj_matrix) <- NA
+
+```
+
+```{r kmodes clustering and graphing, warning= FALSE}
+
+fit2 <- kmodes(DF_class, 5)
+fit2$cluster
+
+g <- graph.adjacency(adj_matrix,mode="undirected")
+
+plot(g,layout=layout.fruchterman.reingold, vertex.size=8, vertex.color=fit2$cluster)
+
+```
+
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
 
-## Please render your code as an .html file using knitr and Pull Resquest both your .Rmd file and .html files to the Assignment 3 repository.
+```{r from assignment 2, warning=FALSE}
+
+#Load data
+DF1 <- read.csv("Data/HUDK405019-clustering.csv", header = TRUE)
+
+#Convert the index numbers of the data frame into the student names.
+
+DF1 <- unite(DF1, "Name", c("First.Name", "Last.Name"), sep = " ")
+
+DF1$Name <- toupper(DF1$Name)
+
+row.names(DF1) <- DF1$Name 
+
+DF1$Name <- NULL
+
+#Wrangle data using dplyr to include only the numerical values.
+
+#Remove location variables
+DF2 <- dplyr::select(DF1, 1:11)
+
+#Remove any characters
+DF2 <- DF2 %>% mutate_all(funs(gsub("[a-zA-Z]", "", .)))
+
+#Convert all variables to numeric
+DF2 <- DF2 %>% mutate_all(funs(as.numeric(.)))
+
+#Scale the data so that no variable has undue influence
+DF2 <- as.data.frame(scale(DF2))
+ 
+#Replace missing values with average score EG - zero
+DF2 <- DF2 %>% mutate_all(funs(ifelse(is.na(.) == TRUE, 0, .)))
+
+fit3 <- kmodes(DF2, 5)
+
+```
+
+```{r}
+
+#Find the students that filled up both surveys. Turns out there are only 32.
+commonstuds <- intersect(rownames(DF1), rownames(DF_class))
+
+DF_class_2 <- DF_class[commonstuds,]
+DF1_2 <- DF1[commonstuds,]
+
+fit4 <- kmodes(DF_class_2, 5)
+fit5 <- kmodes(DF1_2, 5)
+
+overlap <- data.frame(row.names(DF_class_2), fit4$cluster, fit5$cluster)
+row.names(overlap) <- overlap$row.names.DF_class_2.
+overlap$row.names.DF_class_2. <- NULL
+overlap$overlap <- ifelse(overlap$fit4.cluster == overlap$fit5.cluster, 1, 0 )
+
+overlap
```

Comment:
Try to create a data visualization showing an overlap such as a graph or a mosaic plot. Nonetheless, great job overall!